clients/unshaded/pom.xml

```diff
@@ -5,7 +5,6 @@
     <artifactId>tachyon-clients</artifactId>
     <version>0.7.0-SNAPSHOT</version>
   </parent>
-  <groupId>org.tachyonproject</groupId>
```

Comment:
The groupdId of a child pom is by default inherited from its parent pom. Before this PR, in our code some poms have defined the groupId (which is unnecessary anyway), while some other poms haven't. So in this PR, I just cleanup the poms by removing this redundant definitions to keep pom files consistent.